### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by YingboMa**
This PR fixes the following warnings
```julia
[ Info: Precompiling ForwardDiff2 [994df76e-a4c1-5e1f-bd5c-23b9b5303d4f]
WARNING: Method definition rrule(typeof(Base.sum), AbstractArray{#s695, N} where N where #s695<:Real) in module ChainRules at /home/scheme/.julia/dev/ChainRules/src/rulesets/Base/mapreduce.jl:61 overwritten at /home/scheme/.julia/dev/ChainRules/src/rulesets/Base/mapreduce.jl:76.
  ** incremental compilation may be fatally broken for this module **

WARNING: Method definition rrule(typeof(Base.sum), AbstractArray{#s695, N} where N where #s695<:Real) in module ChainRules at /home/scheme/.julia/dev/ChainRules/src/rulesets/Base/mapreduce.jl:61 overwritten at /home/scheme/.julia/dev/ChainRules/src/rulesets/Base/mapreduce.jl:76.
  ** incremental compilation may be fatally broken for this module **
```
and

```julia
julia> using NaNMath, SpecialFunctions
┌ Warning: Error requiring NaNMath from ChainRules:
│ LoadError: UndefVarError: SpecialFunctions not defined
│ Stacktrace:
│  [1] include at ./boot.jl:328 [inlined]
│  [2] include_relative(::Module, ::String) at ./loading.jl:1105
│  [3] include at ./Base.jl:31 [inlined]
│  [4] include(::String) at /home/scheme/.julia/dev/ChainRules/src/ChainRules.jl:1
│  [5] top-level scope at /home/scheme/.julia/dev/ChainRules/src/ChainRules.jl:45
│  [6] eval at ./boot.jl:330 [inlined]
│  [7] eval at /home/scheme/.julia/dev/ChainRules/src/ChainRules.jl:1 [inlined]
│  [8] (::ChainRules.var"#867#873")() at /home/scheme/.julia/packages/Requires/9Jse8/src/require.jl:67
│  [9] err(::ChainRules.var"#867#873", ::Module, ::String) at /home/scheme/.julia/packages/Requires/9Jse8/src/require.jl:38
│  [10] #866 at /home/scheme/.julia/packages/Requires/9Jse8/src/require.jl:66 [inlined]
│  [11] withpath(::ChainRules.var"#866#872", ::String) at /home/scheme/.julia/packages/Requires/9Jse8/src/require.jl:28
│  [12] (::ChainRules.var"#865#871")() at /home/scheme/.julia/packages/Requires/9Jse8/src/require.jl:65
│  [13] #invokelatest#1 at ./essentials.jl:709 [inlined]
│  [14] invokelatest at ./essentials.jl:708 [inlined]
│  [15] #3 at /home/scheme/.julia/packages/Requires/9Jse8/src/require.jl:19 [inlined]
│  [16] iterate at ./generator.jl:47 [inlined]
│  [17] _collect(::Array{Function,1}, ::Base.Generator{Array{Function,1},Requires.var"#3#4"}, ::Base.EltypeUnknown, ::Base.HasShape{1}) at ./array.jl:635
│  [18] map at ./array.jl:564 [inlined]
│  [19] loadpkg(::Base.PkgId) at /home/scheme/.julia/packages/Requires/9Jse8/src/require.jl:19
│  [20] #invokelatest#1 at ./essentials.jl:709 [inlined]
│  [21] invokelatest at ./essentials.jl:708 [inlined]
│  [22] require(::Base.PkgId) at ./loading.jl:925
│  [23] require(::Module, ::Symbol) at ./loading.jl:917
│  [24] eval(::Module, ::Any) at ./boot.jl:330
│  [25] eval_user_input(::Any, ::REPL.REPLBackend) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/REPL/src/REPL.jl:86
│  [26] run_backend(::REPL.REPLBackend) at /home/scheme/.julia/packages/Revise/S7mrl/src/Revise.jl:1057
│  [27] (::Revise.var"#85#87"{REPL.REPLBackend})() at ./task.jl:333
│ in expression starting at /home/scheme/.julia/dev/ChainRules/src/rulesets/packages/NaNMath.jl:4
└ @ Requires ~/.julia/packages/Requires/9Jse8/src/require.jl:40
```

### CODE REVIEW

File: src/rulesets/Base/mapreduce.jl [hunk 1]
```diff
@@ -57,13 +57,6 @@ function frule(::typeof(sum), x)
     return sum(x), sum_pushforward
 end
 
-function rrule(::typeof(sum), x::AbstractArray{<:Real})
```
Comment (mattBrzezinski) [inline src/rulesets/Base/mapreduce.jl]: Is there a reason why this function is being removed? You didn't mention in the CR body.

Comment (mattBrzezinski) [reply src/rulesets/Base/mapreduce.jl]: Nvm, I can see it in your commit message.

**Comment by oxinabox** [comment]
Is there a way to do this without moving like hundreds of lines directly into `__init__`?
What exactly is that solving?
There has to be a better way.

**Comment by YingboMa** [comment]
This is how it is done in JuliaDiffEq https://github.com/JuliaDiffEq/DiffEqBase.jl/blob/master/src/init.jl#L9-L175. I don't know what's bad about removing a few files. If anything, I think it is easier to deal with fewer files since you don't need to keep jumping around files.

**Comment by YingboMa** [comment]
Fine... Seems like that the culprit is https://github.com/MikeInnes/Requires.jl/issues/65

**Comment by oxinabox** [comment]
If this fixes the warning then this LGTM

**Pull request merged**